corehq/apps/userreports/reports/builder/forms.py

```diff
@@ -633,6 +633,17 @@ def _build_data_source(self):
         tasks.rebuild_indicators.delay(data_source_config._id)
         return data_source_config._id
 
+    def _change_indicators(self, indicators):
+        number_cols = [col["field"] for col in self._report_columns if col["aggregation"] in ["avg", "sum"]]
+        changed = False
+        if number_cols:
+            for indicator in indicators:
+                if indicator["column_id"] in number_cols and indicator["datatype"] not in ["integer", "decimal"]:
+                    indicator["datatype"] = "decimal"
```

Comment:
i went with decimal here, because it is impossible to know from a case property whether the original value was an integer or decimal. this only affects case based data sources. form ones retain their original type, so can stay integer or decimal depending on the question